src/migrateCalculatedMeasuresToActivePivot/migrateCalculatedMeasuresInWidgets.test.ts

```diff
@@ -0,0 +1,112 @@
+import { migrateCalculatedMeasuresInWidgets } from "./migrateCalculatedMeasuresInWidgets";
+import { uiWidgetsFolder } from "../__test_resources__/aui5.0LegacyTestResources/uiWidgetsFolder";
+import { dataModelsForTests } from "@activeviam/data-model";
+
+const dataModel = dataModelsForTests.sandbox;
+
+// "pvSum ^ 2" is from cube "EquityDerivativesCubeDist", all others are from "EquityDerivativesCube".
+const namesOfCalculatedMeasurestoMigrate = [
+  "Distinct count city",
+  "Test calculated measure",
+  "EXP pnl.Forex",
+  "Log pv.SUM",
+  "pvSum ^ 2",
+];
+
+describe("migrateCalculatedMeasuresInWidgets", () => {
+  const { cubeNames, migratedWidgetsRecord } =
+    migrateCalculatedMeasuresInWidgets(
+      uiWidgetsFolder,
+      dataModel,
+      namesOfCalculatedMeasurestoMigrate,
+    );
+
+  it("returns a `cubeNames` object containing the names of all calculated measures used in a `ui/widgets` folder with their corresponding cube name", () => {
+    expect(cubeNames).toStrictEqual({
+      "Distinct count city": "EquityDerivativesCube",
+      "EXP pnl.Forex": "EquityDerivativesCube",
+      "Log pv.SUM": "EquityDerivativesCube",
+      "Test calculated measure": "EquityDerivativesCube",
+      "pvSum ^ 2": "EquityDerivativesCubeDist",
+    });
+  });
+
+  it("returns the widget record unchanged if a widget at the root contains no calculated measures", () => {
+    // "7a5" is a widget containing no calculated measures at the root. This is definitely defined, otherwise the snapshot would be empty.
```

Comment:
This is good to justify using `!` to cast as non-nullable in general, but in tests we can be a bit more lenient.
Here in particular, I wouldn't want to give the impression that it's linked to the sentence above.
So either I would move that second sentence below - as close as possible to what it refers to, or just remove it.
Same for other occurrences in the file.

Independently from that, I don't understand "otherwise the snapshot would be empty" => we're not using a snapshot here ?

Reply:
In that case I will just remove the justifications for using `!`. 
It's true that we are not using a snapshot here, that's my mistake, I just copy/pasted "otherwise the snapshot would be empty" from one of the other tests where I used `.toMatchInlineSnapshot()`, but I'm removing this now anyway.